src/movetype.cpp

```diff
@@ -446,12 +446,16 @@ movetype::terrain_info::terrain_info(const terrain_info & that,
 	copy_data(that);
 }
 
+// This is effectively the move constructor for terrain_info, so it's impossible to
+// use 'that' to move construct a terrain_info here.
```

Comment:
I don't follow this comment.

Reply:
Plus, the `NOLINTNEXTLINE` bit is confusing since you *are* moving from `that` for `unique_data_` and `shared_datta_`.

Reply:
I was going to write:

```
// This isn't a real move constructor, because it requires the fallback
// parameter to be explicitly specified. This makes clang-tidy unhappy.
// Normally, this linter check is disabled for move constructors, because
// it would be impossible to std::move a terrain_info in the move
// constructor for terrain_info (we'd move all of the necessary
// member variables instead). We need to explicitly silence the check here.
// NOLINTNEXTLINE(cppcoreguidelines-rvalue-reference-param-not-moved)
```
However, `clang-tidy` has an `AllowPartialMove` option that I'll enable instead. The official C++ core guidelines policy is that if you take something by r-value reference, you must move that thing. However, this is inflexible, as sometimes you'll take an object by r-value reference so you can move individual parts of it. `clang-tidy` has an option to make this guideline allow for cases like this.